funding/manager.go

```diff
@@ -517,6 +539,13 @@ type Manager struct {
 	// goroutine safe.
 	resMtx sync.RWMutex
 
+	// channelFeeMap stores channel fees that were provided to the
+	// reservation context from the initFundingMsg.
+	// When the reservation context is deleted in the FundingSigned
+	// handler then we map the permanent channel id here to the
+	// fees that will be announced in a new channel announcement.
+	channelFeeMap map[lnwire.ChannelID]*htlcswitch.ForwardingPolicy
```

Comment:
As I am unfamiliar with storing fields in buckets in the database I was seeking inspiration from here: https://github.com/lightningnetwork/lnd/blob/d9f4b36cb9009d9e52b952095b612b0edcc3e1e9/funding/manager.go#L4221

Should I follow the example of how the `channelOpeningState` is saved in the DB? Or is there a better example in the code you can point me to on how to store the `channelFeeMap`?

And can we then load the `channelFeeMap` when the funding manager starts? https://github.com/lightningnetwork/lnd/blob/d9f4b36cb9009d9e52b952095b612b0edcc3e1e9/funding/manager.go#L609

Thanks for your help.

Reply:
Also tagging @guggero here for above question.

Reply:
We generally have three options for adding new fields to existing DB objects (with `bbolt` at least).
1. Run a DB migration and add the new field to each existing element in the DB (overkill for this use case IMO because not a lot of users will have pending channels at the time of the `lnd` upgrade and a migration also creates a downgrade barrier which is kind of annoying this early in the 0.16 release schedule).
2. Use TLVs which allows us to add new fields very easily. But since this component doesn't use TLVs yet, switching to TLVs would require a DB migration as well, which we want to avoid.
3. Store new/optional data in a separate key or bucket (depending on the existing structure).

So tl;dr: I agree that we probably should put this field into its own key in `SaveChannelOpeningState` (`channeldb/db.go`).
And while we're at it: I think this property should be stored with each pending channel and not be kept in a global map in the manager. At least I think it makes things easier to understand if the channel knows what fees it should set.
But I haven't taken a closer look at the rest of the PR, so maybe it's not practical to do it in this case? Just drive-by commenting here...

Reply:
Thanks @guggero, this is helpful info I can ponder on.

Re your suggestion about pending channels - this PR proposes to store the fees for pending channels in their respective reservation context. But once the channel is confirmed the reservation context is deleted. Then we need the global map to map the completed channel id to the fees which is then read in the `newChanAnnouncement` - at least as suggested by my PR so far.

Reply:
Thanks @guggero, this is helpful info I can ponder on.
> So tl;dr: I agree that we probably should put this field into its own key in `SaveChannelOpeningState` (`channeldb/db.go`). 

Do you suggest that we probably should create an extra bucket for channel fees per channel_id next to`channelOpeningStateBucket = []byte("channelOpeningState")` and update/delete it with a set of new methods, like `SaveChannelFees` etc.?

>And while we're at it: I think this property should be stored with each pending channel and not be kept in a global map in the manager. At least I think it makes things easier to understand if the channel knows what fees it should set.

This PR proposes to store the fees for pending channels in their respective reservation context. But once the channel is in the 'Funding Signed' stage the reservation context is deleted. Then we need the global map to map the completed channel id to the fees which is then read in the `newChanAnnouncement`. So I guess we could replace the global map by storing the channel fee information in the database before the reservation context is deleted?

Reply:
@guggero, @positiveblue , I've added a new channel fee bucket to the database where fees are persisted and retrieved during the funding process(https://github.com/lightningnetwork/lnd/pull/6753/files#diff-887e88ea126293e649522fa992766c53d32790d424fca8aca4404ad1dc17751d).

I've accompanied my changes with test cases aiming to test fee settings and update and deletion of the db bucket.